File: src/main/java/miragefairy2019/mod/modules/ore/material/IBlockVariantMaterials.java
package miragefairy2019.mod.modules.ore.material;

import miragefairy2019.mod.lib.multi.IBlockVariant;
import net.minecraft.block.SoundType;
import net.minecraft.block.material.Material;

public interface IBlockVariantMaterials extends IBlockVariant {

    public float getBlockHardness();

    public String getHarvestTool();

    public int getHarvestLevel();

    public int getBurnTime();

    public SoundType getSoundType();

    public boolean isFallable();

    public Material getMaterial();

}

File: src/main/java/miragefairy2019/mod/modules/ore/material/IBlockVariantMaterials.kt
package miragefairy2019.mod.modules.ore.material

import miragefairy2019.mod.lib.IBlockVariant
import net.minecraft.block.SoundType
import net.minecraft.block.material.Material

interface IBlockVariantMaterials : IBlockVariant {
    val blockHardness: Float
    val harvestTool: String
    val harvestLevel: Int
    val burnTime: Int
    val soundType: SoundType
    val isFallable: Boolean
    val material: Material?
    val isBeaconBase: Boolean
}
